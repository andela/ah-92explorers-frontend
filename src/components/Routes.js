import React, { Component, Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Signup from './Views/Signup.jsx';
import { Continuee } from './continue/continue';
import Home from './Home.jsx';
import Login from './Containers/Login.jsx';

class Routes extends Component {
  render() {
    return (
      <Router>
        <Switch>
          <Fragment>
            <Route exact path="/" component={Home} />
            <Route path="/signup" component={Signup} />
            <Route path="/continue" component={Continuee} />
            <Route exact path="/login" component={Login} />
          </Fragment>
        </Switch>
      </Router>
    );
  }
}

export default Routes;
