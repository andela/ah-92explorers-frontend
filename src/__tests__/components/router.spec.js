import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../../components/Routes';

describe('Route Component', () => {
  it('should render Routes component', () => {
    const component = shallow(<Routes />);
    expect(component).toMatchSnapshot();
  });
});
