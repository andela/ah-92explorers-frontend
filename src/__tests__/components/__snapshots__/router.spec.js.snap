// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Component should render Routes component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Home)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Signup)",
          "type": [Function],
        }
      }
      path="/signup"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Continue)",
          "type": [Function],
        }
      }
      path="/continue"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/login"
    />
  </Switch>
</BrowserRouter>
`;
